app/uk/gov/hmrc/agentsubscription/controllers/RegistrationController.scala

```diff
@@ -18,24 +18,28 @@ package uk.gov.hmrc.agentsubscription.controllers
 
 import javax.inject._
 
+import com.kenshoo.play.metrics.Metrics
 import play.api.libs.json.Json
 import play.api.libs.json.Json.toJson
-import play.api.mvc.{Action, AnyContent, Result}
-import uk.gov.hmrc.agentsubscription.auth.{AuthActions, RequestWithAuthority}
-import uk.gov.hmrc.agentsubscription.connectors.AuthConnector
-import uk.gov.hmrc.agentsubscription.model.{postcodeWithoutSpacesRegex}
-import uk.gov.hmrc.agentmtdidentifiers.model.{Utr}
+import play.api.mvc.{AnyContent, Request, Result}
+import uk.gov.hmrc.agentsubscription.connectors.{AuthConnector, Provider}
+import uk.gov.hmrc.agentsubscription.model.postcodeWithoutSpacesRegex
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentsubscription.MicroserviceAuthConnector
 import uk.gov.hmrc.agentsubscription.service.RegistrationService
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
 import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.Future
 
 @Singleton
-class RegistrationController @Inject()(service: RegistrationService, override val authConnector: AuthConnector)
-  extends BaseController with AuthActions {
+class RegistrationController @Inject()(service: RegistrationService)
+                                      (implicit metrics: Metrics, microserviceAuthConnector: MicroserviceAuthConnector)
+  extends AuthConnector(metrics, microserviceAuthConnector) with BaseController {
 
-  private[controllers] def getRegistrationBlock(utr: Utr, postcode: String): RequestWithAuthority[AnyContent] => Future[Result] = { implicit request =>
+  private[controllers] def getRegistrationBlock(utr: Utr, postcode: String)
```

Comment:
I think we can find a better name than `getRegistrationBlock` or simply inline this function if is applied only once?